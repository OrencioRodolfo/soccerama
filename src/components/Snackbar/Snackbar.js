import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from 'material-ui/styles'
import Snackbar from 'material-ui/Snackbar'
import IconButton from 'material-ui/IconButton'
import CloseIcon from 'material-ui-icons/Close'

const styles = theme => ({
  close: {
    width: theme.spacing.unit * 4,
    height: theme.spacing.unit * 4,
  },
})

class CustomSnackbar extends React.Component {
  static propTypes = {
    message: PropTypes.string.isRequired,
    onClose: PropTypes.func,
    open: PropTypes.bool,
  }

  static defaultProps = {
    onClose: () => {},
    open: false,
  }

  handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return
    }

    this.props.onClose()
  }

  render() {
    return (
      <Snackbar
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'right',
        }}
        open={this.props.open}
        autoHideDuration={6000}
        onClose={this.handleClose}
        SnackbarContentProps={{
          'aria-describedby': 'message-id',
        }}
        message={<span id="message-id">{this.props.message}</span>}
        action={[
          <IconButton
            key="close"
            className="icon-close"
            aria-label="Close"
            color="inherit"
            onClick={this.handleClose}
          >
            <CloseIcon />
          </IconButton>,
        ]}
      />
    )
  }
}

export default withStyles(styles)(CustomSnackbar)
export {
  CustomSnackbar as DCustomSnackbar,
}
