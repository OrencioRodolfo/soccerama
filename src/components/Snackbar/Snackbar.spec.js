import React from 'react'
import { shallow } from 'enzyme'
import { DCustomSnackbar } from './Snackbar'

describe('Global error component', () => {
  let wrapper
  let onClose

  beforeEach(() => {
    onClose = jest.fn()
    wrapper = shallow(<DCustomSnackbar
      message="sample"
      onClose={onClose}
    />)
  })

  it('must render properly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('must render properly with default props', () => {
    expect(<DCustomSnackbar
      message="sample"
    />).toMatchSnapshot()
  })

  it('must do nothing', () => {
    const result = wrapper.instance().handleClose(undefined, 'clickaway')
    expect(result).toEqual(undefined)
  })

  it('must call the onClose callback', () => {
    wrapper.instance().handleClose()
    expect(onClose).toBeCalled()
  })
})
