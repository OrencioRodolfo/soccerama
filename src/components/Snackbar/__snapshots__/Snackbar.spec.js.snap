// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global error component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomSnackbar
    message="sample"
    onClose={[MockFunction]}
    open={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "SnackbarContentProps": Object {
        "aria-describedby": "message-id",
      },
      "action": Array [
        <WithStyles(IconButton)
          aria-label="Close"
          className="icon-close"
          color="inherit"
          onClick={[Function]}
        >
          <pure(Close) />
        </WithStyles(IconButton)>,
      ],
      "anchorOrigin": Object {
        "horizontal": "right",
        "vertical": "top",
      },
      "autoHideDuration": 6000,
      "message": <span
        id="message-id"
      >
        sample
      </span>,
      "onClose": [Function],
      "open": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "SnackbarContentProps": Object {
          "aria-describedby": "message-id",
        },
        "action": Array [
          <WithStyles(IconButton)
            aria-label="Close"
            className="icon-close"
            color="inherit"
            onClick={[Function]}
          >
            <pure(Close) />
          </WithStyles(IconButton)>,
        ],
        "anchorOrigin": Object {
          "horizontal": "right",
          "vertical": "top",
        },
        "autoHideDuration": 6000,
        "message": <span
          id="message-id"
        >
          sample
        </span>,
        "onClose": [Function],
        "open": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
